ibm/resource_ibm_satellite_host.go

```diff
@@ -0,0 +1,322 @@
+// Copyright IBM Corp. 2017, 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package ibm
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/IBM-Cloud/bluemix-go/bmxerror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.ibm.com/ibmcloud/kubernetesservice-go-sdk/kubernetesserviceapiv1"
+)
+
+const (
+	hostCluster    = "cluster"
+	hostLocation   = "location"
+	hostName       = "host_id"
+	hostLabels     = "labels"
+	hostZone       = "zone"
+	hostWorkerPool = "worker_pool"
+	hostProvider   = "host_provider"
+
+	rsHostNormalStatus       = "normal"
+	rsHostProvisioningStatus = "provisioning"
+	rsHostReadyStatus        = "ready"
+	rsHostUnknownStatus      = "unknown"
+)
+
+func resourceIBMSatelliteHost() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMSatelliteHostCreate,
+		Read:     resourceIBMSatelliteHostRead,
+		Update:   resourceIBMSatelliteHostUpdate,
+		Delete:   resourceIBMSatelliteHostDelete,
+		Exists:   resourceIBMSatelliteHostExists,
+		Importer: &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(45 * time.Minute),
+			Update: schema.DefaultTimeout(45 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			hostCluster: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name or ID of a Satellite location or cluster to assign the host to",
```

Comment:
hostCluster, hostLocation attributes should have Optional and exactlyoneof =true